File: Math - 1/C. Почти простое число.py
n = int(input())
i = 2
divs = []
while i * i <= n:
    while n % i == 0:
        divs.append(i)
        n //= i
    i += 1
if n > 1:
    divs.append(n)
 
print("YES" if len(divs) == 2 and divs[0] != divs[1] else "NO")

File: Math - 1/C. Почти простое число.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    long long n; cin >> n;
    long long x = n;
    vector<long long>divs;

    for (int i = 2; i * i <= x; i++)
        while (x % i == 0) {
            divs.push_back(i);
            x /= i;
        }
    if (x > 1)
        divs.push_back(x);
    cout << ((divs.size() == 2 && divs[0] != divs[1]) ? "YES" : "NO");
}
